src/Controllers/SourceController.php

```diff
@@ -63,6 +63,13 @@ class SourceController extends Controller
      */
     public function add(Source $source, $mode = HttpHandler::MODE_EXECUTE)
     {
+        $endpoint = $source->getEndpoint();
+        // For Klarna, use a different different endpoint for production
+        if (strpos($endpoint, 'klarna') !== false) {
+            if(!$this->getSandbox()) {
+                $endpoint = 'klarna/credit-sessions';
+            }
+        }
```

Comment:
Hey @ioan-ghisoi-cko ,
Ideally you want to override `getEndpoint` in Klarna class. `SourceController.php` should be agnostic to any source.